### ISSUE TITLE
Update cellxgene to work with h5py>=3.0.0

### ISSUE THREAD

**Issue by bmccandless** [chanzuckerberg/cellxgene#1963]
h5py returns back bytes instead of str for string data.   This breaks a few assumptions which causes the server to fail on certain inputs.  Many tests are failing.
As a short term fix, a requirement for h5py<3.0.0 has been introduced.
For the future we should migrate cellxgene to use the new behavior (or see if there are any options in h5py to maintain the current behavior).  

Also update the compatibility tests to exclude the anndata version that isn't compatible.

**Comment by ivirshup** [comment]
Update from the `anndata` side of things, `anndata>=0.7.5` handles the `h5py` string handling changes.

**Comment by bmccandless** [comment]
I'm not sure there is a better solution to this problem than what we currently have.
It isn't ideal, but not causing any problem. 
 
Currently:
anndata >= 0.7.0
h5py < 3.0.0 # added recently because 3.0.0 introduced a breaking change.

anndata was updated in 0.7.5 to work with the new version of h5py

The issue:
if anndata < 0.7.5, then we must set h5py < 3.0.0, or the code fails.
if anndata >= 0.7.5, then no need to set the h5py version.

The problem is that there is not a way to express this in the requirements.txt format.

Solutions:
1. require anndata >= 0.7.5 going forward, and remove h5py requirement.

2. leave the requirements as is, meaning that h5py will remain at < 3.0.0.  
If we even need to bump our minimum anndata version to 0.7.5 or higher, then we can remove the h5py requirement.
So, basically option 1, but at a later date.

3. modify cellxgene to work with older versions of anndata, but new version of h5py.   It could be done, but I'd argue that it's not the place to solve this problem.  It would require a bit of work for cellxgene with limited benefit.

**Comment by maniarathi** [comment]
cc @mckinsel but option 2 seems reasonable to me. If we get to a state where we _need_ to support h5py >3.0.0, then we can revisit it and close this for now. What do you think?

**Comment by ivirshup** [comment]
@bmccandless Could you share the issue you're running into with anndata 0.7.5 and h5py < 2.10.0? We aim to support that.

**Comment by bmccandless** [comment]
Hi @ivirshup,  

There is no issue, as far as I know, with anndata==0.7.5 and h5py<2.10.0.
Thanks for fixing it in anndata.

This particular issue for cellxgene can be solved by requiring anndata>=0.7.5.
However, we were trying to give flexibility to our users around which anndata version they were using.
So, if a user had anndata<0.7.5 then we want to make sure that continues to work.
The solution so far has been to require h5py<3.0.0.

**Comment by ivirshup** [comment]
Ah yes, I misread this before having my coffee 😊, nvm.
